cmd/root.go

```diff
@@ -127,6 +127,10 @@ any client SSL certificates.`,
 	cmd.PersistentFlags().Uint64Var(&c.conf.MaxConnections, "max-connections", 0,
 		`Limits the number of connections by refusing any additional connections.
 When this flag is not set, there is no limit.`)
+	cmd.PersistentFlags().DurationVar(&c.conf.WaitOnClose, "wait-after-sigterm", 0,
+		`Amount of time to wait for any open connections to close before
```

Comment:
Nit: consider "max-sigterm-delay"

Maybe change the wording to clarify the behavior as well:
"Maximum amount of time to wait after for any open connections after received a sigterm signal. The proxy will shutdown when open connections hits 0 or when the max time has passed. Defaults to 0s."

Reply:
That's a better name. I've slightly tweaked the description.